src/catalog/system_catalogs.cpp

```diff
@@ -106,6 +107,16 @@ void SystemCatalogs::Bootstrap(const std::string &database_name,
   if (!pg_query_metrics_) {
     pg_query_metrics_ = new QueryMetricsCatalog(database_name, txn);
   }
+
+  // Reset oid of each catalog to avoid collisions between catalog
+  // values added by system and users when checkpoint recovery.
+	pg_attribute_->UpdateOid(OID_FOR_USER_OFFSET);
+	pg_namespace_->UpdateOid(OID_FOR_USER_OFFSET);
+	pg_table_->UpdateOid(OID_FOR_USER_OFFSET);
+	pg_index_->UpdateOid(OID_FOR_USER_OFFSET);
+	// pg_layout_->UpdateOid(OID_FOR_USER_OFFSET);
+	pg_trigger_->UpdateOid(OID_FOR_USER_OFFSET);
+	// pg_proc->UpdateOid(OID_FOR_USER_OFFSET);
```

Comment:
This is not dead code because the pg_proc would be implemented within the system catalog according to this file's line 95-97. To avoid to forget this function call, it also keeps comment out.